### DISCUSSION TITLE
Filtered Squirrel and Chameleon datasets

### DISCUSSION THREAD

**Discussion by j-mesnil** [pyg-team/pytorch_geometric#7531] [Ideas]
In a [recent paper](https://arxiv.org/abs/2302.11640), a number of issues were found with the Squirrel and Chameleon datasets, which can be found in the WikipediaNetwork dataset class in PyG. They however provide [reworked versions](https://github.com/yandex-research/heterophilous-graphs/tree/main/data) of said datasets which alleviate some of these issues.

Given that the original datasets proposed by the authors have already made their way into the PyG dataset collection, would it be possible to add the filtered versions of the Squirrel and Chameleon datasets as well? Perhaps providing a switch in the WikipediaNetwork dataset class could be an option for that.

Thank you!

**Comment by rusty1s** [comment]
Thanks for letting us know. I am very open to this idea. Do you wanna help us with this effort?

**Comment by j-mesnil** [comment]
Surely! I hacked together a version for my own use which seems to work well, I based it on the existing dataset class of the WikipediaNetwork. On the weekend I will clean it up and make a pull request for it.

What would be the preferred option for adding the dataset? Should I add a boolean switch (similar to the existing `geom_gcn_preprocess`)? Also, there are directed and undirected versions of the dataset provided by the authors of the recent paper, which one should be used here preferably?

**Reply by rusty1s** [reply]
Thank you <3. At best, we can introduce this via a boolean switch) to avoid code duplication.

**Comment by MicheleCattaneo** [comment]
Hi there,
Have these issues been fixed for the WikipediaNetwork datasets? I could not find anything in closed PRs and commit histories.
Thank you!

**Reply by j-mesnil** [reply]
Hi! I'm sorry, I lost track of publishing the changes I made, so there haven't been any changes from my side.
